src/main/java/gregtech/api/recipes/ingredients/GTRecipeFluidInput.java

```diff
@@ -74,23 +96,13 @@ public boolean equals(Object obj) {
             return false;
         }
         GTRecipeFluidInput other = (GTRecipeFluidInput) obj;
-        if (this.amount != other.amount) return false;
-        if (this.isConsumable != other.isConsumable) return false;
-        if (this.nbtMatcher != null && !this.nbtMatcher.equals(other.nbtMatcher)) return false;
-        if (this.nbtCondition != null && !this.nbtCondition.equals(other.nbtCondition)) return false;
-        if (inputStack.getFluid().getName().equals(other.inputStack.getFluid().getName())) {
-            return FluidStack.areFluidStackTagsEqual(this.inputStack, other.inputStack);
-        }
-        return false;
-    }
 
-    @Override
-    public int hashCode() {
-        if (nbtMatcher == null) {
-            //the Fluid registered to the fluidName on game load might not be the same Fluid after loading the world, but will still have the same fluidName.
-            return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, inputStack.tag);
-        }
-        return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, 0);
+        if (this.amount != other.amount || this.isConsumable != other.isConsumable) return false;
+        if (!Objects.equals(this.nbtMatcher, other.nbtMatcher)) return false;
+        if (!Objects.equals(this.nbtCondition, other.nbtCondition)) return false;
+
+        return areFluidsEqual(this.inputStack, other.inputStack) &&
```

Comment:
Does `other.inputStack` need to be null checked?

Reply:
I think `inputStack` field should be expected to be present.